inc/task.class.php

```diff
@@ -1749,128 +1783,150 @@ public static function getItemsFromDB($filter)
     {
         global $DB;
 
-        $select = ["tasks" => "task.*"];
-        $where = [];
-        $leftjoin = [];
+        $criteria = [
+            'SELECT' => ['task.*'],
+            'FROM' => 'glpi_plugin_glpiinventory_tasks AS task',
+            'WHERE' => []
+        ];
 
-       // Filter active tasks
+        // Filter active tasks
         if (
             isset($filter['is_active'])
               and is_bool($filter['is_active'])
         ) {
-            $where[] = "task.`is_active` = " . $filter['is_active'];
+            $criteria['WHERE'][] = ['task.is_active' => $filter['is_active']];
         }
 
-       //Filter by running taskjobs
+        //Filter by running taskjobs
         if (
             isset($filter['is_running'])
               and is_bool($filter['is_running'])
         ) {
-           // add taskjobs table JOIN statement if not already set
-            if (!isset($leftjoin['taskjobs'])) {
-                $leftjoin_bak = $leftjoin;
-                $leftjoin_tmp = PluginGlpiinventoryTaskjob::getJoinQuery();
-                $leftjoin = array_merge($leftjoin_bak, $leftjoin_tmp);
-                if (!isset($select["taskjobs"])) {
-                    $select['taskjobs'] = "taskjob.*";
-                }
+            // add taskjobs table JOIN statement
+            if (!isset($criteria['LEFT JOIN'])) {
+                $criteria['SELECT'] = array_merge(
+                    $criteria['SELECT'],
+                    [
+                        'taskjob.id AS taskjob_id',
+                        'taskjob.plugin_glpiinventory_tasks_id AS taskjob_plugin_glpiinventory_tasks_id',
+                        'taskjob.entities_id AS taskjob_entities_id',
+                        'taskjob.name AS taskjob_name',
+                        'taskjob.date_creation AS taskjob_date_creation',
+                        'taskjob.method AS taskjob_method',
+                        'taskjob.targets AS taskjob_targets',
+                        'taskjob.actors AS taskjob_actors',
+                        'taskjob.comment AS taskjob_comment',
+                        'taskjob.rescheduled_taskjob_id AS taskjob_rescheduled_taskjob_id',
+                        'taskjob.statuscomments AS taskjob_statuscomments',
+                        'taskjob.enduser AS taskjob_enduser',
+                        'taskjob.restrict_to_task_entity AS taskjob_restrict_to_task_entity'
+                    ]
+                );
+                $criteria['LEFT JOIN'] = PluginGlpiinventoryTaskjob::getJoinCriteria();
```

Comment:
IMHO, it would be easier to read to have the JOIN directives written here directly.